src/store/index.ts

```
@@ -80,41 +90,103 @@ export default createStore<State>({
         payload: messageData.ciphertext
       });
     },
-    async uploadFile({ dispatch }, request: UploadRequest) {
+    async uploadFile({ dispatch, getters, state }, request: UploadRequest) {
       const res = await axios.post<UploadResponse>(
         "https://rtc-data.registree.io/data/uploads",
         request
       );
-      if (res.status !== 200) {
+      const enclavePubKey = getters.enclavePublicKey;
+      const ourSecretKey = state.ourSecretKey;
+
+      if (res.status !== 200 || !enclavePubKey || !ourSecretKey) {
+        // TODO: Error Handling
+        return "error";
```

We probably want to make all these placeholder error cases use `throw`, rather than `return`?